src/main/java/org/aesh/parser/LineParser.java

```diff
@@ -198,7 +197,6 @@ else if (parseCurlyAndSquareBrackets && (c == CURLY_END || c == PARENTHESIS_END)
                     handleCurlyEnd(c);
                 }
                 else if (haveEscape) {
-                    builder.append(BACK_SLASH);
```

Comment:
Same comment. I would love ;-), to have the 2 parsing operations be merged in order to avoid to duplicate fixes.

Reply:
we should probably look at trying to refactor that method so it could be inlined...

Reply:
we should probably look at trying to refactor that method so it could be inlined...

Reply:
@MMarus , I will try to spend some time (once your fixed is merged) to refactor the 2 methods.

Reply:
I have incorporated the changes you suggested into the PR. I didn't refactor the 2 methods in order to keep the PR cleaner.